types/src/verifier/mod.rs

```diff
@@ -45,6 +53,70 @@ pub trait ProofVerifier {
         }
     }
 
+    fn verify_aggregate_seals(
+        aggregate: &AggregateSealVerifyProofAndInfos,
+    ) -> Result<(), Box<dyn StdError>> {
+        if aggregate.infos.is_empty() {
+            return Err("no seal verify infos".into());
+        }
+        let spt: proofs::RegisteredSealProof = aggregate.seal_proof.try_into()?;
+        let prover_id = prover_id_from_u64(aggregate.miner);
+
+        struct AggregationInputs {
+            commr: [u8; 32],
+            commd: [u8; 32],
+            sector_id: u64,
+            ticket: [u8; 32],
+            seed: [u8; 32],
+        }
+        let inputs: Vec<AggregationInputs> = aggregate
+            .infos
+            .iter()
+            .map(|info| {
+                let commr = cid_to_replica_commitment_v1(&info.sealed_cid)?;
+                let commd = cid_to_data_commitment_v1(&info.unsealed_cid)?;
+                Ok(AggregationInputs {
+                    commr,
+                    commd,
+                    sector_id: info.sector_number,
+                    ticket: info.randomness.0,
+                    seed: info.interactive_randomness.0,
+                })
+            })
+            .collect::<Result<Vec<_>, &'static str>>()?;
+
+        let inp: Vec<Vec<_>> = inputs
+            .par_iter()
+            .map(|input| {
+                seal::get_seal_inputs(
+                    spt,
+                    input.commr,
+                    input.commd,
+                    prover_id,
+                    input.sector_id.try_into()?,
+                    input.ticket,
+                    input.seed,
+                )
+            })
+            .try_reduce(Vec::new, |mut acc, current| {
+                acc.extend(current);
+                Ok(acc)
+            })?;
+        let commrs: Vec<[u8; 32]> = inputs.iter().map(|input| input.commr).collect();
+        let seeds: Vec<[u8; 32]> = inputs.iter().map(|input| input.seed).collect();
+        if !verify_aggregate_seal_commit_proofs(
+            spt,
+            aggregate.aggregate_proof.try_into()?,
+            aggregate.proof.clone(),
+            &commrs,
+            &seeds,
+            inp,
+        )? {
+            Err("Invalid Aggregate Seal proof".into())
```

Comment:
Upsetting that this doesn't use a custom error type. Would it be worth it to create a ticket to define the error types here? Do you anticipate any complications will come up?

Reply:
This trait is fairly small.

Reply:
I dont see why its useful or important here. These methods are generally used in actors. And the way the error is handled is irrespective of what error happens here anyways. I think a refactor of error handling in this might mess up with error propagation in actors.